src/chart_plugin/systems/arrows.rs

```diff
@@ -98,27 +98,18 @@ pub fn redraw_arrows(
     mut arrow_query: Query<(Entity, &ArrowMeta), With<ArrowMeta>>,
     mut commands: Commands,
 ) {
-    let mut despawned: HashSet<ArrowMeta> = HashSet::new();
-
     for event in redraw_arrow.iter() {
         for (entity, arrow) in &mut arrow_query.iter_mut() {
-            if despawned.contains(arrow) {
-                continue;
-            }
             if arrow.start.id == event.id || arrow.end.id == event.id {
-                if let Some(entity) = commands.get_entity(entity) {
-                    despawned.insert(*arrow);
-                    entity.despawn_recursive();
+                if let Some(mut entity) = commands.get_entity(entity) {
+                    entity.despawn();
+                    create_arrow.send(CreateArrow {
+                        start: arrow.start,
+                        end: arrow.end,
+                        arrow_type: arrow.arrow_type,
+                    });
```

Comment:
This refactoring not gonna work... if you run with code changes in this file application hangs with
`Could not despawn entity 46283v0 because it doesn't exist in this World.`

I suggest to return the previous version of the code for this file.